src/pick.c

```diff
@@ -463,6 +470,7 @@ selected_choice(void)
 			filter_choices();
 			selection = 0;
 		}
+		yscroll = 0;
```

Comment:
Nice catch! I ended up begin explicit and resetting `yscroll` when
appropriate. See ab2fe6f.